app/controllers/GuidanceController.scala

```diff
@@ -41,33 +43,47 @@ class GuidanceController @Inject() (
 ) extends FrontendController(mcc)
     with I18nSupport {
 
+  val logger = Logger(getClass)
+
   def getPage(path: String): Action[AnyContent] = Action.async { implicit request =>
-    withSession(service.getPageContext(s"/$path", _)).map {
-      case Some(pageContext) =>
-        Logger.info(s"Retrieved page at ${pageContext.page.urlPath}, start at ${pageContext.processStartUrl}")
+    withSession[PageContext](service.getPageContext(s"/$path", _)).map {
+      case Right(pageContext) =>
+        logger.info(s"Retrieved page at ${pageContext.page.urlPath}, start at ${pageContext.processStartUrl}")
         pageContext.page match {
           case page: StandardPage => Ok(standardView(page, pageContext.processStartUrl))
           case page: QuestionPage => Ok(questionView(page, pageContext.processStartUrl, questionName(path), formProvider(questionName(path))))
         }
-      case None =>
-        Logger.warn(s"Request for PageContext at /$path returned nothing resulting in BadRequest")
+      case Left(NotFoundError) =>
+        logger.warn(s"Request for PageContext at /$path returned NotFound, returning NotFound")
+        NotFound(errorHandler.notFoundTemplate)
+      case badRequesterr @ Left(BadRequestError | InvalidProcessError) =>
```

Comment:
Looking at the service method getPageContext I think it can raise InvalidProcessError and NotFound. Further, the repository can raise DatabaseError. As such I don't think we need to test for BadRequest. An InvalidProcessError would suggest something is wrong with the process returned from Mongo so perhaps we should return InternalServerError.

Reply:
Being called without a sessionId will pass through this route., this seems to be a valid BadRequestError